pylint/checkers/typecheck.py

```diff
@@ -806,6 +806,10 @@ def _is_invalid_isinstance_type(arg: nodes.NodeNG) -> bool:
         return False
     if isinstance(inferred, astroid.Instance) and inferred.qname() == BUILTIN_TUPLE:
         return False
+    if isinstance(arg, nodes.BinOp) and arg.op == "|":
```

Comment:
Can't we use the new `UnionType` here?

Reply:
Because of what I assume are inference failures, I couldn't cover all the test cases without checking both the inferred value and the superficial value. Is there a better way to do this?